source/game/system/map/MapdataArea.cc

```diff
@@ -0,0 +1,178 @@
+#include "MapdataArea.hh"
+
+#include "game/system/CourseMap.hh"
+
+namespace System {
+
+MapdataAreaBase::MapdataAreaBase(const SData *data, s16 index) : m_rawData(data), m_index(index) {
+    EGG::RamStream stream = EGG::RamStream(data,
+            CourseMap::Instance()->version() > 2200 ? sizeof(SData) : sizeof(SData) - 4);
+    read(stream);
+
+    m_index = -1;
+    m_sqBoundingSphereRadius = 0.0f;
+    m_ellipseAspectRatio = 0.0f;
+    m_ellipseRadiusSq = 0.0f;
+    m_dimensions = EGG::Vector3f::zero;
+    m_right = EGG::Vector3f::zero;
+    m_up = EGG::Vector3f::zero;
+    m_forward = EGG::Vector3f::zero;
+}
+
+void MapdataAreaBase::read(EGG::Stream &stream) {
+    m_shape = static_cast<Shape>(stream.read_s8());
+    m_type = static_cast<Type>(stream.read_s8());
+    m_cameraIdx = stream.read_s8();
+    m_priority = stream.read_u8();
+    m_position.read(stream);
+    m_rotation.read(stream);
+    m_scale.read(stream);
+
+    for (auto &param : m_params) {
+        param = stream.read_s16();
+    }
+
+    if (CourseMap::Instance()->version() > 2200) {
+        m_railId = stream.read_s8();
+        m_enemyId = stream.read_u8();
+    }
+}
+
+/// @addr{0x80516168}
+MapdataPointInfo *MapdataAreaBase::getPointInfo() const {
+    // The rail ID doesn't exist in prior versions, so there's no point info
+    auto *courseMap = CourseMap::Instance();
+    if (courseMap->version() < 2200) {
+        return nullptr;
+    }
+
+    return courseMap->getPointInfo(m_railId);
+}
+
+MapdataAreaBox::MapdataAreaBox(const SData *data, s16 index) : MapdataAreaBase(data, index) {
+    m_dimensions.x = 0.5f * (10000.0f * m_scale.x);
+    m_dimensions.y = 10000.0f * m_scale.y;
+    m_dimensions.z = 0.5f * (10000.0f * m_scale.z);
+
+    m_ellipseAspectRatio = 0.0f;
+    m_ellipseRadiusSq = 0.0f;
+    m_sqBoundingSphereRadius = m_dimensions.squaredLength();
+
+    EGG::Quatf rotation = EGG::Quatf::FromRPY(DEG2RAD * m_rotation.x, DEG2RAD * m_rotation.y,
+            DEG2RAD * m_rotation.z);
```

Comment:
Not sure if we have the operator overload to do multiplication on the lhs
```suggestion
    EGG::Quatf rotation = DEG2RAD * m_rotation;
```

Reply:
We have the friend operator, but it resolves to Vector3f * float, which is not our intended behavior here.